packages/embeds/embed-core/src/embed.ts

```diff
@@ -305,27 +305,34 @@ export class Cal {
     //     },
     //   },
     // });
-    let attributesString = "";
-    let existingEl = null;
+    let existingEl: HTMLElement | null = null;
+
     if (attributes?.id) {
-      attributesString += ` id="${attributes.id}"`;
       existingEl = document.getElementById(attributes.id);
     }
-    let el = existingEl;
+    let el: HTMLElement;
     if (!existingEl) {
       const template = document.createElement("template");
-      template.innerHTML = `<cal-floating-button ${attributesString}  data-cal-namespace="${this.namespace}" data-cal-link="${calLink}"></cal-floating-button>`;
```

Comment:
There was a reason to use the `innerHTML` method to create the custom element cal-floating-button. Using the createElement method doesn't actually make it work due to some reason. I don't remember the reason now :D. Can you test it once, it will break certainly. Floating button popup won't come

Reply:
Nevermind. I tested it and it was failing. I have pushed a fix for it.

Reply:
Also, e2e were passing. It means that we don't have floatingButton popup tests. Need to add those.